### DISCUSSION TITLE
Broadcast transaction from wallet

### DISCUSSION THREAD

**Discussion by sebastianbermea** [cosmos/community#76] [Clients (JS)]
Hello,

Im trying to broadcast a transaction using StargateClient but with DirectSignResponse (window,keplr) signatures, is it possible?
When i use client.broadcastTx() i get invalid request. Error code 18

```
const signature = decodeSignature(directResponse.signature);
    const tx = TxRaw.fromPartial({
      bodyBytes: unSigned.bodyBytes,
      authInfoBytes: unSigned.authInfoBytes,
      signatures: [signature],
    });
    const txByte = TxRaw.encode(tx).finish();
    const client = await StargateClient.connect(RPC_URL);
    const result = await client.broadcastTx(
      txByte
    );
```

**Comment by sebastianbermea** [comment]
> Hello,
> 
> Im trying to broadcast a transaction using StargateClient but with DirectSignResponse (window,keplr) signatures, is it possible? When i use client.broadcastTx() i get invalid request. Error code 18
> 
> ```
> const signature = decodeSignature(directResponse.signature);
>     const tx = TxRaw.fromPartial({
>       bodyBytes: unSigned.bodyBytes,
>       authInfoBytes: unSigned.authInfoBytes,
>       signatures: [signature],
>     });
>     const txByte = TxRaw.encode(tx).finish();
>     const client = await StargateClient.connect(RPC_URL);
>     const result = await client.broadcastTx(
>       txByte
>     );
> ```

I found the error for invalid request but now im getting Error code 2 tx parse error

**Reply by tac0turtle** [reply]
there should be an error attached to the log , what does it say there?
